xdsl/builder.py

```diff
@@ -107,6 +127,113 @@ def region(
         else:
             return Builder._region_args(input)
 
+    @staticmethod
+    def _implicit_region_no_args(func: Callable[[], None]) -> Region:
+        """
+        Generates a single-block region.
+        """
+        block = Block()
+        builder = Builder(block)
+
+        with Builder._push_implicit_builder(builder):
+            func()
+
+        return Region.from_block_list([block])
+
+    @staticmethod
+    def _implicit_region_args(
+        input_types: list[Attribute] | ArrayAttr[Attribute]
+    ) -> Callable[[_CallableImplicitRegionFuncType], Region]:
+        """
+        Decorator for constructing a single-block region, containing the implementation of a
+        function with some input arguments.
+        """
+
+        if isinstance(input_types, ArrayAttr):
+            input_type_seq = input_types.data
+        else:
+            input_type_seq = input_types
```

Comment:
```suggestion
        if isinstance(input_types, ArrayAttr):
            input_types = input_types.data
```
I usually find this more concise, and this is enough for Pyright

Reply:
It wasn't enough for my version locally, the type was still a union with ArrayAttr, annoyingly